src/WinSparkleDotNet/WinSparkle.cs

```diff
@@ -0,0 +1,557 @@
+        ///     </para>
+        /// </remarks>
+        ///
+        /// <param name="intervalSeconds">   The interval in seconds between checks for updates.
+        ///                     The minimum update interval is 3600 seconds (1 hour). </param>
+        [DllImport("WinSparkle.dll", EntryPoint = "win_sparkle_set_update_check_interval", CharSet = CharSet.Ansi, CallingConvention = CallingConvention.Cdecl)]
+        private static extern void SetUpdateCheckInterval(
+            [param: MarshalAs(UnmanagedType.I4)] int intervalSeconds
+            );
+
+        /// <summary>   
+        ///     <para>
+        ///         Gets the automatic update interval in seconds.
+        ///     </para>
+        ///     <para>
+        ///         Default value is one day.
+        ///     </para>
+        /// </summary>
+        ///
+        /// <remarks>
+        ///     <para>
+        ///         Can only be called @em before the first call to <see cref="Initialize"/>
+        ///     </para>
+        /// </remarks>
+        ///
+        /// <returns>   The update check interval in seconds. </returns>
+        [DllImport("WinSparkle.dll", EntryPoint = "win_sparkle_get_update_check_interval", CallingConvention = CallingConvention.Cdecl)]
+        [return: MarshalAs(UnmanagedType.I4)]
+        private static extern int GetUpdateCheckInterval();
+
+        /// <summary>   
+        ///     <para>
+        ///         Gets the time for the last update check.
+        ///     </para>
+        ///     <para>
+        ///         Default value is -1, indicating that the update check has never run.
+        ///     </para>
+        /// </summary>
+        ///
+        /// <remarks>
+        ///     <para>
+        ///         Can only be called @em before the first call to <see cref="Initialize"/>
+        ///     </para>
+        /// </remarks>
+        ///
+        /// <returns>   The last check time internal. </returns>
+        [DllImport("WinSparkle.dll", EntryPoint = "win_sparkle_get_last_check_time", CallingConvention = CallingConvention.Cdecl)]
+        [return: MarshalAs(UnmanagedType.I4)]
+        private static extern int GetLastCheckTime();
+
+        [UnmanagedFunctionPointer(CallingConvention.StdCall)]
+        public delegate void ErrorCallback();
+
+        /// <summary>   
+        ///     <para>
+        ///         Set callback to be called when the updater encounters an error.
+        ///     </para>
+        /// </summary>
+        /// <param name="callback"> The callback. </param>
+        [DllImport("WinSparkle.dll", EntryPoint = "win_sparkle_set_error_callback", CharSet = CharSet.Ansi, CallingConvention = CallingConvention.Cdecl)]
+        public static extern void SetErrorCallback(
+            [param: MarshalAs(UnmanagedType.FunctionPtr)] ErrorCallback callback
+            );
+
+        [UnmanagedFunctionPointer(CallingConvention.StdCall)]
+        public delegate bool CanShutdownCallback();
```

Comment:
The same applies to the other delegate declarations.

How about that `bool` vs. `int` return type. Will that be marshalled correctly?